src/ui/components/AppWrapper/hooks/useActivityTimer.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useRef, useState } from "react";
+import { App } from "@capacitor/app";
+import { useAppDispatch } from "../../../../store/hooks";
+import { logout } from "../../../../store/reducers/stateCache";
+
+const timeout = process.env.NODE_ENV === "development" ? 3600000 : 60000; // 1h/1min
+const pauseTimeout = timeout / 2;
+const useActivityTimer = () => {
+  const dispatch = useAppDispatch();
+  const [pauseTimestamp, setPauseTimestamp] = useState(0);
+  const timer = useRef<NodeJS.Timeout | null>(null);
+
+  const clearTimer = () => {
+    if (timer.current) {
+      clearTimeout(timer.current);
+    }
+  };
+
+  const setActivityTimer = () => {
+    clearTimer();
+    timer.current = setTimeout(() => {
+      dispatch(logout());
+    }, timeout);
+  };
+
+  const handleActivity = () => {
+    setActivityTimer();
+  };
+
+  useEffect(() => {
+    const pauseListener = App.addListener("pause", () => {
+      const now = new Date().getTime();
+      setPauseTimestamp(now);
+    });
+
+    const resumeListener = App.addListener("resume", () => {
+      const now = new Date().getTime();
+      if (now - pauseTimestamp > pauseTimeout) {
+        dispatch(logout());
+      }
```

Comment:
So this check runs when the app comes back in - if I re-open the app after 40 seconds, does it appear unlocked and then "jump" close then, or its so fast that I only see the lock screen?

Reply:
Yes, it appears unlocked for a few milisec then we see the lock screen

Reply:
You could argue that putting the app in and out of the background is an "action". So when it gets put to the background we could clear the timer and call `setTimeout` with 30 seconds, and when it comes back in do the same but with 60 seconds.

This would avoid the UI glitchy feeling

Reply:
I tried it, but when the app is the background it freezes, and no operation can be performed, so I had to save the pause time and then check it was more than 30sec

Reply:
Hmm, what happens then if the app is in the background for more than 60 seconds? If that logs out correctly, I'm not sure why it would freeze if we are just clearing and re-creating the timeout

Reply:
would be the same behavior as if 30 seconds had passed.
When returning to the app you would see for a few millisec the app unlocked.
If you had a counter inside the pause event, you will se how the counter will not add up until you bring the app back to the front

Reply:
Oh so when its in the background the timeouts stop working completely?

Reply:
yes, it stops.
To avoid seeing the app unlocked for a few secs before login page.. when `pause` event we can show a loading page, then on `resume`, hide loading page and then show the login page

Reply:
for me it's OK if that happens, we should just let the team know and see if there's any strong opinions on it. But it mightn't be worth complicating things